framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C

```diff
@@ -41,15 +47,104 @@ MultiAppVariableValueSamplePostprocessorTransfer::validParams()
+  if (_directions.contains("from_multiapp"))
+  {
+    // Check that the variable is a CONSTANT MONOMIAL.
+    auto & fe_type = _var.feType();
+    if (fe_type.order != CONSTANT || fe_type.family != MONOMIAL)
+      paramError("source_variable",
+                 "Variable must be in CONSTANT MONOMIAL when transferring from a postprocessor "
+                 "from sub-apps.");
+
+    if (!_fe_problem.getAuxiliarySystem().hasVariable(_var_name))
+      paramError("source_variable", "Variable must be an auxiliary variable");
+  }
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::setupPostprocessorCommunication()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  const auto num_global_apps = getFromMultiApp()->numGlobalApps();
+
+  // Setup the communication pattern
+  _postprocessor_to_processor_id.resize(num_global_apps,
+                                        std::numeric_limits<processor_id_type>::max());
+  for (const auto i : make_range(num_global_apps))
+    if (getFromMultiApp()->hasLocalApp(i))
+      _postprocessor_to_processor_id[i] = this->processor_id();
+
+  _communicator.min(_postprocessor_to_processor_id);
+#ifdef DEBUG
+  for (const auto i : make_range(num_global_apps))
+  {
+    mooseAssert(_postprocessor_to_processor_id[i] != std::numeric_limits<processor_id_type>::max(),
+                "Every element in the vector should have been set.");
+    if (getFromMultiApp()->hasLocalApp(i))
+      mooseAssert(_postprocessor_to_processor_id[i] == this->processor_id(),
+                  "If I owned this app, then the processor id value should be my own");
+  }
+#endif
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::cacheElemToPostprocessorData()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  // Cache the Multiapp position ID for every element.
+  auto & mesh = _fe_problem.mesh().getMesh();
+  unsigned int multiapp_pos_id = 0;
+  for (auto & elem : as_range(mesh.active_local_elements_begin(), mesh.active_local_elements_end()))
+    // Exclude the elements without dofs.
+    if (_var.hasBlocks(elem->subdomain_id()))
+    {
+      Real distance = std::numeric_limits<Real>::max();
+      unsigned int count = 0;
+      for (unsigned int j = 0; j < getFromMultiApp()->numGlobalApps(); ++j)
+      {
+        Real current_distance = (getFromMultiApp()->position(j) - elem->true_centroid()).norm();
+        if (MooseUtils::absoluteFuzzyLessThan(current_distance, distance))
+        {
+          distance = current_distance;
+          multiapp_pos_id = j;
+          count = 0;
+        }
+        else if (MooseUtils::absoluteFuzzyEqual(current_distance, distance))
+          ++count;
+      }
+      if (count > 0)
+        mooseError("The distances of an element to more than one sub-applications are too close."
+                   "\nDifferent positions for sub-applications or a centroid-based MultiApp can "
+                   "be used to resolve this error.");
```

Comment:
```suggestion
                   "\nThe mapping from postprocessors to variables is not unique. The element with the lowest ID was chosen. Different positions for sub-applications or a centroid-based MultiApp can "
                   "be used to resolve this indetermination.");
```

Reply:
a good candidate for a mooseWarningOnce but this isnt ready yet

Reply:
I think there is a misunderstanding here. We have more candidate sub-apps providing the postprocessor values for an element. We must set values for all elements but now cannot choose one unless we choose the one with smaller sub-app id. I still think issuing an error is better than this bandage to make things deterministic. I can add here that users can choose to block restrict the variable to fix this.

Reply:
smaller subapp id seems fine then. I must have missed that the for loop is on subapps not elements